client/sites-dashboard-v2/index.tsx

```diff
@@ -129,6 +133,12 @@ const SitesDashboardV2 = ( {
 		window.setTimeout( () => updateQueryParams( queryParams ) );
 	}, [ dataViewsState.search, statusSlug, updateQueryParams ] );
 
+	// Update URL with page param on change.
+	useEffect( () => {
+		const queryParams = { page: dataViewsState.page };
+		window.setTimeout( () => updateQueryParams( queryParams ) );
+	}, [ dataViewsState.page, updateQueryParams ] );
```

Comment:
If we have the query param for page set to anything other than 1 and reload the window, it resets to 1 still. This makes keeping track of it in the URL seem pointless. Im not 100% on why this happens, but wondering if its due to something in the initialization of the underlying gutenberg components.